File: src/main/scala/com/worldmanager/megamind/api/repositories/BaseMongoRepository.scala
package com.worldmanager.megamind.api.repositories

import java.io.Serializable

import com.worldmanager.megamind.api.models.Entity
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.repository.NoRepositoryBean

@NoRepositoryBean
trait BaseMongoRepository[T <: Entity, ID <: Serializable] extends MongoRepository[T, ID]

File: src/main/java/com/example/demo/repository/BaseMongoRepository.java
package com.example.demo.repository;

import com.example.demo.domain.Entity;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.repository.NoRepositoryBean;

@NoRepositoryBean
public interface BaseMongoRepository<T extends Entity, ID> extends MongoRepository<T, ID> {

}
